dessia_common/report.py

```diff
@@ -0,0 +1,199 @@
+        self.to_txt()
+
+    def add_title(self, title: str):
+        self.last_offset = 0
+        title = title.upper()
+        lines = ['# ' + title]
+
+        self.add_lines(lines, nb_line_blank=2)
+
+    def add_subtitle(self, title: str):
+        self.last_offset = 0
+        title = title[0].upper() + title[1:]
+        if len(title) > self.width_line:
+            title = title[0: self.width_line]
+
+        lines = ['', '## ' + title]
+        self.add_lines(lines, nb_line_blank=1)
+
+    def add_subsubtitle(self, title: str):
+        offset = 2
+        self.last_offset = offset
+        title = title[0].upper() + title[1:]
+        if len(title) > self.width_line - offset:
+            title = title[0: self.width_line - offset]
+
+        lines = [' ' * offset + '### ' + title]
+        self.add_lines(lines, offset, nb_line_blank=0)
+
+    def add_text(self, text: str):
+        offset = self.last_offset + 2
+        line_length = self.width_line - offset
+
+        nb_bloc = text.split(' ')
+        lines = []
+        line = ' ' * offset
+        for bloc in nb_bloc:
+            if bloc == "\n":
+                lines.append(line)
+                line = ' ' * offset
+
+            elif len(line) + len(bloc) <= line_length:
+                line += bloc
+                if len(line) + 1 <= line_length:
+                    line += ' '
+            else:
+                if len(line) == offset:
+                    line += bloc
+                    lines.append(line)
+                    line = ' ' * offset
+                else:
+                    lines.append(line)
+                    line = ' ' * offset + bloc + ' '
+
+        lines.append(line)
+        self.add_lines(lines, offset)
+
+    def add_table(self, title: List[str], elements: List[List[any]]):
+
+        # | Left columns  | Right columns |
+        # | ------------- |:-------------:|
+        # | left foo      | right foo     |
+        # | left bar      | right bar     |
+        # | left baz      | right baz     |
+
+        offset = self.last_offset + 2
+        # line_length = self.width_line - offset
+
+        real_line_length = self.width_line - offset - (len(title) + 1) - 2 * len(title)
+        if real_line_length < 0:
+            raise KeyError('too many columns in the report')
+
+        max_length = [len(t) for t in title]
+
+        for element in elements:
+            for index, elem in enumerate(element):
+                max_length[index] = max(max_length[index], len(str(elem)))
+
+        pourcent_length = [m / sum(max_length) for m in max_length]
+
+        real_length = []
+        for pourcent in pourcent_length[0:-1]:
+            real_length.append(int(pourcent * real_line_length))
+        real_length.append(real_line_length - sum(real_length))
+
+        lines = ['  \n  ']
+        lines.append(self.add_table_line(offset, real_length, title))
+        lines.append(self.add_table_line(offset, real_length, title, is_title = True))
```

Comment:
I think one method with a parameter like that is a good thing, there are less line in the code at the end
If i code two methods it would be a duplicate